manifests/init.pp

```diff
@@ -0,0 +1,130 @@
+#
+# @param create_mysql_home
+# Should the mysql_home be created?
+#
+# @param dbs
+# Raw data to define mysql::db resources.
+#
+# @param mysql_gid
+# The GID for the mysql user.
+#
+# @param mysql_groupname
+# The name of the mysql group.
+#
+# @param mysql_home
+# The home dir of the mysql user.
+#
+# @param mysql_logdir
+# Location of MySQL log files.
+#
+# @param mysql_parentdir
+# Location of directory that contains the mysql_home.
+# NOTE: Might be useful for this to be an array so multiple nested
+# dirs can be created as dependencies.
+#
+# @param mysql_rundir
+# Location of run directory (generally a sub-directory of /var/run/).
+#
+# @param mysql_uid
+# UID of the mysql user.
+#
+# @param mysql_username
+# Username of the mysql user.
+#
+# @param other_dependencies
+# Other depencies to handle prior to other MySQL setup, specified as resources, e.g.:
+#   - "Mount['/var/lib/mysql']"
+#
+# @param yumrepo
+# Raw params for a yumrepo resource from which to install MySQL/MariaDB.
+#
+# @example
+# include profile_mysql_server
+#
+class profile_mysql_server (
+
+  Boolean           $create_mysql_home,
+  Hash              $dbs,
+  Integer           $mysql_gid,
+  String            $mysql_groupname,
+  String            $mysql_home,
+  String            $mysql_logdir,
+  Optional[String]  $mysql_parentdir,
+  String            $mysql_rundir,
+  Integer           $mysql_uid,
+  String            $mysql_username,
+  Array             $other_dependencies,
+  Hash              $yumrepo,
+
+) {
+
+  include ::mysql::server
+
+  $dir_defaults = {
+    before => Class['::mysql::server::install'],
+    ensure => directory,
+    group  => $mysql_gid,
+    owner  => $mysql_uid,
+  }
+
+  if $mysql_logdir {
+    file { $mysql_logdir:
+      * => $dir_defaults,
+    }
+  }
+  if $mysql_rundir {
+    file { $mysql_rundir:
+      * => $dir_defaults,
+    }
+  }
+  if $mysql_parentdir {
+    file { $mysql_parentdir:
+      * => $dir_defaults,
+    }
+  }
+
+  if $create_mysql_home {
+    file { $mysql_home:
+      * => $dir_defaults,
+    }
+  }
+
+  $other_dependencies.each | $dep | {
+    $dep -> Class['::mysql::server::install']
+  }
+
+  group { $mysql_groupname:
+    ensure => 'present',
+    before => Class['::mysql::server::install'],
+    gid    => $mysql_gid,
+  }
+
+  user { $mysql_username:
+    ensure         => 'present',
+    before         => Class['::mysql::server::install'],
+    uid            => $mysql_uid,
+    gid            => $mysql_gid,
+    forcelocal     => true,
+    home           => $mysql_home,
+    managehome     => $create_mysql_home,
+    password       => '!!',
+    purge_ssh_keys => true,
+    shell          => '/sbin/nologin',
+    comment        => 'MySQL server',
+  }
+
+  if ! empty($yumrepo) {
+    yumrepo { 'mysql_yumrepo':
+      before => Class['::mysql::server::install'],
+      *      => $yumrepo,
+    }
+  }
+
+  each($dbs) | $db_name, $db_data | {
+    mysql::db { $db_name:
+      * => $db_data,
+    }
+  }
```

Comment:
Again, perhaps a place where `ensure_resources (‘mysql::db’, $dbs)` might make more sense here? 
The main need for this is if other modules may also create dbs (which they probably are not currently).

Reply:
I guess I'm not too concerned about this, because I think we'd only have a conflict if another module tried to manage a DB with the exact same name, right?